runtime/lua/vim/lsp/codelens.lua

```diff
@@ -31,10 +31,24 @@ local function execute_lens(lens, bufnr, client_id)
   local line = lens.range.start.line
   api.nvim_buf_clear_namespace(bufnr, namespaces[client_id], line, line + 1)
 
+  local command = lens.command
+  local fn = vim.lsp.commands[command.command]
+  if fn then
+    fn(command, { bufnr = bufnr, client_id = client_id })
+    return
+  end
   -- Need to use the client that returned the lens → must not use buf_request
   local client = vim.lsp.get_client_by_id(client_id)
   assert(client, 'Client is required to execute lens, client_id=' .. client_id)
-  client.request('workspace/executeCommand', lens.command, function(...)
+  local command_provider = client.server_capabilities.executeCommandProvider
```

Comment:
Also I still dislike this pattern of sanitizing capabilities and think we should change it but it's consistent with everything else so not important.